### PR TITLE
Loading and error states for cluster templates and repositories tables

### PR DESCRIPTION
**Pull request by jtomasek**
Screenshot 2022-09-30 at 11 31 38
Screenshot 2022-09-30 at 11 33 59
Screenshot 2022-09-30 at 11 56 38

Signed-off-by: Jiri Tomasek <jtomasek@redhat.com>

### CODE REVIEW

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx [hunk 1]
```diff
@@ -202,37 +195,44 @@ export const RowContext = React.createContext<{
...
+        errorMessage={t('The Helm repositories could not be loaded.')}
       >
-        <TableComposable
-          aria-label="Helm repositories table"
-          id="helm-repositories-table"
-          variant="compact"
+        <RowContext.Provider
+          value={{
+            helmRepoIndexResult: [repoIndex, repoIndexLoaded, repoIndexError],
+            clusterTemplatesResult: useClusterTemplates(),
```
Comment (rawagner) [inline frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx:213]: lets move the hook call somewhere higher

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx [hunk 2]
```diff
@@ -202,37 +195,44 @@ export const RowContext = React.createContext<{
...
+        loaded={repositoriesLoaded}
+        error={repositoriesError}
+        errorId="helm-repositories-load-error"
+        errorMessage={t('The Helm repositories could not be loaded.')}
       >
-        <TableComposable
-          aria-label="Helm repositories table"
-          id="helm-repositories-table"
-          variant="compact"
+        <RowContext.Provider
```
Comment (rawagner) [inline frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx:210]: I think we dont need the context anymore, with the composable table we can easily just pass these as props to rows

**Review by rawagner — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx (hunk 2 above)
Comment (jtomasek) [reply frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx]: done

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx (hunk 1 above)
Comment (jtomasek) [reply frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplatesPage/HelmRepositoriesTab.tsx]: done

**Pull request merged**